modules/open8BidAdapter.js

```diff
@@ -0,0 +1,182 @@
+import { Renderer } from 'src/Renderer';
+import {ajax} from '../src/ajax';
+import * as utils from 'src/utils';
+import { registerBidder } from 'src/adapters/bidderFactory';
+import { VIDEO, BANNER } from 'src/mediaTypes';
+
+const BIDDER_CODE = 'open8';
+const URL = '//as.vt.open8.com/v1/control/prebid';
+const AD_TYPE = {
+  VIDEO: 1,
+  BANNER: 2
+};
+
+export const spec = {
+  code: BIDDER_CODE,
+  supportedMediaTypes: [VIDEO, BANNER],
+
+  isBidRequestValid: function(bid) {
+    return !!(bid.params.slotKey);
+  },
+
+  buildRequests: function(validBidRequests, bidderRequest) {
+    var requests = [];
+    for (var i = 0; i < validBidRequests.length; i++) {
+      var bid = validBidRequests[i];
+      var queryString = '';
+      var slotKey = utils.getBidIdParameter('slotKey', bid.params);
+      queryString = utils.tryAppendQueryString(queryString, 'slot_key', slotKey);
+      queryString = utils.tryAppendQueryString(queryString, 'imp_id', generateImpId());
+      queryString = utils.tryAppendQueryString(queryString, 'bid_id', bid.bidId);
+
+      requests.push({
+        method: 'GET',
+        url: URL,
+        data: queryString
+      });
+    }
+    return requests;
+  },
+
+  interpretResponse: function(serverResponse, request) {
+    var bidderResponse = serverResponse.body;
+
+    if (!bidderResponse.isAdReturn) {
+      return [];
+    }
+
+    var ad = bidderResponse.ad;
+
+    const bid = {
+      slotKey: bidderResponse.slotKey,
+      userId: bidderResponse.userId,
+      impId: bidderResponse.impId,
+      media: bidderResponse.media,
+      ds: ad.ds,
+      spd: ad.spd,
+      fa: ad.fa,
+      pr: ad.pr,
+      mr: ad.mr,
+      nurl: ad.nurl,
+      requestId: ad.bidId,
+      cpm: ad.price,
+      creativeId: ad.creativeId,
+      dealId: ad.dealId,
+      currency: ad.currency || 'JPY',
+      netRevenue: true,
+      ttl: 360, // 6 minutes
+    }
+
+    if (ad.adType === AD_TYPE.VIDEO) {
+      const videoAd = bidderResponse.ad.video;
+      Object.assign(bid, {
+        vastXml: videoAd.vastXml,
+        width: videoAd.w,
+        height: videoAd.h,
+        renderer: newRenderer(bidderResponse),
+        adResponse: bidderResponse,
+        mediaType: VIDEO
+      });
+    } else if (ad.adType === AD_TYPE.BANNER) {
+      const bannerAd = bidderResponse.ad.banner;
+      Object.assign(bid, {
+        width: bannerAd.w,
+        height: bannerAd.h,
+        ad: bannerAd.adm,
+        mediaType: BANNER
+      });
+      try {
+        bannerAd.imps.forEach(impTrackUrl => {
```

Comment:
Could add a check to verify the `bannerAd.imps` is populated before running the `forEach`.